assets/content/cookbook/Other/adding-static-methods-to-existing-classes.md

```diff
@@ -0,0 +1,47 @@
+[tags]: / "static-extensions"
+
+# Adding Static Methods to Existing Classes
+
+Haxe allows you to add static methods to existing classes (eg. `sys.io.File`) via the Static Extensions feature. The "secret sauce" is to specify the first parameter of the extension as `Class<X>` where `X` is the class you want to add static methods to (eg. `sys.io.File`).
+
+Here's a class with a static method that adds a `ensureDirectoryExists` method to `sys.FileSystem`:
+
+## Implementation
+
+```haxe
+// FileSystemExtensions.hx
+class FileSystemExtensions {
+  public static function ensureDirExists(clazz:Class<sys.FileSystem>, path:String) : Void {
+    if (!FileSystem.exists(path)) {
+      throw path + " doesn't exist";
+    }
+    else if (!FileSystem.isDirectory(path)) {
+      throw path + " isn't a directory";
+    }
+  }
+}
+```
+
+Note that the method is `public`, `static`, and the first parameter is `Class<sys.FileSystem>`.
+
+## Usage
+
+Here's how to consume the code:
+
+```haxe
```

Comment:
Yes, I pretty much copy/pasted the code from a working snippet at `try.haxe.org`. Maybe I wasn't careful enough. What problem did you spot?